zep_python/langchain/history.py

```diff
@@ -0,0 +1,169 @@
+    """
+
+    def __init__(
+        self,
+        session_id: str,
+        zep_client: Optional[ZepClient] = None,
+        api_url: Optional[str] = "https://api.getzep.com",
+        api_key: Optional[str] = None,
+    ) -> None:
+        try:
+            import langchain_core.messages.base  # noqa: F401
+        except ImportError:
+            raise ImportError(
+                "Could not import langchain-core package. "
+                "Please install it with `pip install langchain-core`."
+            )
+        if zep_client is None and api_key is None and api_url is None:
+            raise ValueError("Either zep_client or api_key must be provided.")
+
+        if zep_client is None:
+            self.zep_client = ZepClient(api_url=api_url, api_key=api_key)
+        else:
+            self.zep_client = zep_client
+
+        self.session_id = session_id
+
+    @property
+    def messages(self) -> List[BaseMessage]:  # type: ignore
+        """Retrieve messages from Zep memory"""
+
+        from langchain_core.messages import (
+            AIMessage,
+            BaseMessage,
+            HumanMessage,
+            SystemMessage,
+        )
+
+        zep_memory: Optional[Memory] = self._get_memory()
+        if not zep_memory:
+            return []
+
+        messages: List[BaseMessage] = []
+        # Extract summary, if present, and messages
+        if zep_memory.summary:
+            if len(zep_memory.summary.content) > 0:
+                messages.append(SystemMessage(content=zep_memory.summary.content))
+        if zep_memory.messages:
+            msg: Message
+            for msg in zep_memory.messages:
+                metadata: Dict = {
+                    "uuid": msg.uuid,
+                    "created_at": msg.created_at,
+                    "token_count": msg.token_count,
+                    "metadata": msg.metadata,
+                }
+                if msg.role == "ai":
+                    messages.append(
+                        AIMessage(content=msg.content, additional_kwargs=metadata)
+                    )
+                else:
+                    messages.append(
+                        HumanMessage(content=msg.content, additional_kwargs=metadata)
+                    )
+
+        return messages
+
+    @property
+    def zep_messages(self) -> Union[List[Message], None]:
+        """Retrieve summary from Zep memory"""
+        zep_memory: Optional[Memory] = self._get_memory()
+        if not zep_memory:
+            return []
+
+        return zep_memory.messages
+
+    @property
+    def zep_summary(self) -> Optional[str]:
+        """Retrieve summary from Zep memory"""
+        zep_memory: Optional[Memory] = self._get_memory()
+        if not zep_memory or not zep_memory.summary:
+            return None
+
+        return zep_memory.summary.content
+
+    def _get_memory(self) -> Optional[Memory]:
+        """Retrieve memory from Zep"""
+        from zep_python import NotFoundError
```

Comment:
I think you do this on L#6

Reply:
thanks - vestige of having been in the langchain-community package.